velox/expression/CastExpr-inl.h

```diff
@@ -293,6 +293,47 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType,
+    const TypePtr& toType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  auto flatResult = result->asFlatVector<StringView>();
+
+  // Calculate the total size of the stringBuffers.
+  size_t totalResultBytes = 0;
+  applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
+    if (!simpleInput->isNullAt(row)) {
+      std::string output =
+          DecimalUtil::toString((int128_t)simpleInput->valueAt(row), fromType);
```

Comment:
We can more easier to get the length of input value, fromType scale indicates if there is dot, input value  < 0 means there is a `-`, all we can do is to get the 10dec of the value.
https://lemire.me/blog/2021/05/28/computing-the-number-of-digits-of-an-integer-quickly/
https://stackoverflow.com/questions/1489830/efficient-way-to-determine-number-of-digits-in-an-integer

Reply:
https://stackoverflow.com/questions/61707231/why-is-log-base-10-used-in-this-code-to-convert-int-to-string

Reply:
This part was removed because it can be handled with existing method at [FlatVector.cpp#L263-L277](https://github.com/facebookincubator/velox/blob/main/velox/vector/FlatVector.cpp#L263-L277).